blah.cpp

```diff
@@ -0,0 +1,82 @@
+#include <stdio.h>
+#include <stdbool.h>
+#include <iostream>
+#include <cstring>
+#include <cstdlib>
+#include <cctype.h>
+
+bool characteristic(char numString[], int& c);
+bool mantissa(char numString[], int& numerator, int& denominator);
+
+int main()
+{
+    char number[] = "123.456";
+    int c, n, d;
+
+    //if the conversion from C string to integers can take place
+    if(characteristic(number, c) && mantissa(number, n, d))
+    {
+        //do some math with c, n, and d
+        printf("test true \n");
+        printf("The characteristic is: %i\n", c);
+        printf("The mantissa numerator is: %i\n", n);
+
+    }
+    // handles error on input
+    else
+    {
+        int num_plus = 0;
+        // Checks if there is a letter, cover imaginary numbers or random letter chars
+        if(isalpha(number))
+        {      
+               printf("test false \n");
+        }
+        
+        // Another solution
+        // Checks if there is another symbol besides a digit or a + - symbol
+        for (int i = 0; i < strlen(number); i = i + 1)
+        { 
+          if(number[i] >= '0' || number[i] <= '9' && number[i] != '+' && number[i] != '-')
+          {
+                printf("test false");
+          }
+        }
+        return 0; 
+    }
+}    
+    
+bool characteristic(char numString[], int& c)
+{
+    //turn char into int
+    c = atoi(numString);
+    //printf("%i", c);
+    return c;
+}
+
+bool mantissa(char numString[], int& numerator, int& denominator)
+{
+    char mantissa_value_string[100];
+    char mantissa_buffer[100];
+    int currentIndex = 0;
+    int offset = 0;
+
+    while (numString[currentIndex] != '.')
+    {
+        mantissa_buffer[currentIndex] = numString[currentIndex];
+        currentIndex++;
+    }
+
+    currentIndex++;
+    offset = currentIndex;
+
+    while (numString[currentIndex] != '\0')
+    {
+      mantissa_value_string[currentIndex - offset] = numString[currentIndex];
+      currentIndex++;
+    }
+
+    numerator = atoi(mantissa_value_string);
+    //printf("The mantissa numerator is: %i\n", numerator);
+
+    return numerator;

```

Comment:
bool mantissa() returns type int* and not bool?

Reply:
Just to clarify this a bit further, it seems like C++ is similar to C in that bools don't actually exist. It's just an int that holds either 0 or 1. It also seems like the bool can, maybe implicitly, be converted to a number.

Given the above, it seems like in the line
```numerator = atoi(mantissa_value_string);```
This assigns the output to the reference parameter numerator, which acts kind of like a pointer in C, but with more restrictions on it, as C++ is designed(as far as I know) to de-emphasize pointer manipulation that is rampant in C. For this function, this is how the values are returned out of the function, and how the denominator is also suppose to be returned.

Now moving to 
```return numerator;```
What's happening here is that you're trying to return the reference parameter out of a function with the bool return type. This should error, but what seems to be happening is that the return dereferences the numerator and looks at the value it holds, which is an int. It takes the int and tries to fit it in a bool, which is just an int that can only be 0 or 1, thus the number gets reduced down to 1 which is then returned.

I recommend refactoring this code so that the return is a separate variable that solely determines if the function error'd or not, and just passing out the numerator and denominator values through the reference parameters. This also applies to the characteristic function.